### ISSUE TITLE
[Feature]: Allow all transports for image pull in RestApi/libpod

### ISSUE THREAD

**Issue by mwopfner** [containers/podman#17324]
### Feature request description

Currently if the RestApi/libpod is used there is a check which only allows to pull images via the docker transport (https://github.com/containers/podman/blob/main/pkg/api/handlers/libpod/images_pull.go#L54-L58).
This is also documented in the method description (_"Other transports are rejected as they do not make sense in a remote context."_). In general I agree but in my use case it would be good to be not limited by this.

**My use case is:**
I have podman running on an embedded system and a other process checks if a usb stick is plugged into the system. If so it checks if there are new containers on the usb stick and then instruct podman (via RestApi) to pull the containers via the oci transport.

### Suggest potential solution

The simplest solution is to just remove the check (I did this and tested it with the `oci` transport, which just worked).
Maybe a better solution would be to check if it is a local connection and only allow non docker transports than.

### Have you considered any alternatives?

It would also be possible execute (`os.Exec()`) the `podman pull oci:/...` command directly but this would be a mix of `libpod` and direct calls.

### Additional context

I think other locally running tools (like `podman-tui`) which uses the RestApi should be able to use all transports during a pull.

**Comment by vrothberg** [comment]
Thanks for reaching out and sharing your use case, @mwopfner!

The problem with the non-docker transports is a certain ambiguity of their sources.  Does a `podman-remote pull oci:/tmp/foo` relate to `/tmp/foo` on the client or on the host?

Your solution of running it directly on the device is also what I'd recommend.  Maybe automate it with a systemd service that checks for new images on a regular base?

**Comment by vrothberg** [comment]
@mwopfner I sympathize with the idea but I fear it's too restricted to open up the possibility.  The most robust way is to exec `/usr/bin/podman`.  An alternative in Go would be to use `https://github.com/containers/common/tree/main/libimage` as Podman does internally.  We do our best to keep the API of `libimage` stable but there's no guarantee at thepresent.

**Comment by mwopfner** [comment]
I can see the arguments (#11593 is related) but it is really unfortunately because half of the use cases (commands run on the remote) would work but are blocked by the check. But I also understand that the interface is not clear for the user in this case, so disabling it completely makes also sense.

- The only solution to this would be an extra argument which specifies if the path is meant for `client` or `remote` and if a transport is not usable in one context and error could be returned. 
This is a check that `podman-remote` on the client should perform an probably nothing the `RestApi` on the `remote` must check. But to be compatible with older `podman-remote` versions I guess there should still be a check on the `remote` and an extra parameter which enables the other transports in the context of the `remote`.

- I think using `/usr/bin/podman pull` for now is the best option for me to get it working.

**Comment by rhatdan** [comment]
I don't see this as easily being solved. and I am not sure how it work in remote mode with the files on the client side.

**Comment by mwopfner** [comment]
@rhatdan: I thing your correct. Getting this working properly (so that the user also know how to use it correctly) is not easy. I guess technically the most functions should be available to implement it (`podman scp` already allows to copy images from local to remote) but getting the cli interface correct is tedious.

**Comment by mwopfner** [comment]
@vrothberg What I want to achieve is to load an oci archive which is present on the remote (plugged in usb stick) via the rest API (see: UseCase in the first post). So I do not think extending `podman-remote load` will help me here.

**Comment by vrothberg** [comment]
Thanks for the reminder.  For a moment, I forgot about the server-side requirement.

**Comment by mwopfner** [comment]
No problem :wink:. I have a workaround, so this is not blocking me, but it would be nice to have a proper solution for this ... :thinking: 

Actually I think, maybe I could use `podman-remote load` in my case, because I'm using the Rest API from a process running on the remote, so I can also access the usb stick from this "remote" process. 
It is probably not the most efficient solution, but it should work.

**Comment by mwopfner** [comment]
After reading the `podman-remote load` docs and testing it it turns out that it does not work, because I use `oci-dir` which is currently not supported. But seeing `podman-remote load` got me thinking that the only thing (except for remote support for `oci-dir` and `docker-dir`) missing for `podman --remote pull` is to specify the context in which the pull should operate.
Maybe a `--context=local`, `--context=remote` could be enough.

`podman --remote pull --context=local oci-archive:~/alpine.tar` pulls it from the oci tar in my home folder on this machine
`podman --remote pull --context=remote oci:/mnt/usb-stick/alpine` pulls it from the oci folder `/mnt/usb-stick/alpine` on the remote

During testing the API I also run `podman --remote pull skopeo` which showed:
```
Resolved "skopeo" as an alias (/etc/containers/registries.conf.d/000-shortnames.conf)
Trying to pull quay.io/skopeo/stable:latest...
```
but I could not find any explanation in the docs if `000-shortnames.conf` is used from my pc or the remote one.

**Comment by rhatdan** [comment]
@vrothberg what should we do with this one?

**Comment by vrothberg** [comment]
I do not think Podman should support loading archives on the server-side.  This would somehow break (a bit) the client/server model of podman-remote.

**Comment by rhatdan** [comment]
I agree.

**Comment by rhatdan** [comment]
@mwopfner Are you interested in working on this?  @vrothberg Do you think we should close?

**Comment by vrothberg** [comment]
As mentioned above, I don't think remote should support it.
